tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -570,6 +572,44 @@ public function testAlterTableScenario()
         }
     }
 
+
+    public function testAlterTableInNamespace()
+    {
+        if (! $this->_sm->getDatabasePlatform()->supportsSchemas()) {
+            $this->markTestSkipped('Schema definition is not supported by this platform.');
+        }
+
+        if (! in_array('testschema', $this->_sm->listNamespaceNames(), true)) {
+            $diff                  = new SchemaDiff();
+            $diff->newNamespaces[] = 'testschema';
+
+            foreach ($diff->toSql($this->_sm->getDatabasePlatform()) as $sql) {
+                $this->_conn->exec($sql);
+            }
+        }
+
+        $myTable = $this->createTestTable('testschema.my_table');
+        self::assertTrue(in_array('testschema.my_table', $this->_sm->listTableNames(), true));
+
```

Comment:
After this point, the rest of test could be removed as not really relevant to the fix. Covering `dbo` and non-`dbo` tables will be enough to test the changes.